### ISSUE TITLE
"reload" makes all VIPs of vrrp instances switches

### ISSUE THREAD

**Issue by SimonLiuhd** [acassen/keepalived#2159]
**Describe the bug**
reloading the keepalived make all VIPs of all vrrp instances drops

**To Reproduce**
1. all the vips of the vrrp instances exist the machine A (while the backup machine B has no VIPs)
2. change the keepalived.conf to remove a vrrp instance 
3. execute "systemctl reload keepalived"
4. check the vips

**Expected behavior**
only the vip of the removed vrrp instance is drop, and other instances' vips still keeps on machine A

**Keepalived version**
2.2.7

**Distro (please complete the following information):**
 - Name: centos
 - Version: 7.9
 - Architecture: x86_64

**Details of any containerisation or hosted service (e.g. AWS)**
normal vm on vmare

**keepalived.conf**
```
! Configuration File for keepalived

global_defs {
   router_id LVS_DEVEL
   vrrp_skip_check_adv_addr
   vrrp_garp_interval 0
   vrrp_gna_interval 0
}

vrrp_script check_nginx {
    script "/opt/check_nginx.sh"         
    interval 5                                              
    weight  20                                              
    timeout 4
}

vrrp_script check_rest {
    script "/opt/check_rest.sh" 
    interval 5
    weight  20
    timeout 4
}


vrrp_instance nginx {
    state BACKUP
    interface eth0
    virtual_router_id 102
    priority 100
    advert_int 1
    authentication {
        auth_type PASS
        auth_pass password123
    }
   
    track_script {
        check_nginx       
    }
    
    unicast_src_ip 192.168.1.100
    unicast_peer {
        192.168.1.101
    }
    
    virtual_ipaddress {
        192.168.1.102
    }
    
    notify_master    /opt/nginx_status_master.sh
    notify_backup    /opt/nginx_status_backup.sh
}

vrrp_instance rest {
    state BACKUP 
    interface eth0
    virtual_router_id 103
    priority 100
    advert_int 1
    authentication {
        auth_type PASS
        auth_pass password123
    }
   
    track_script {
        check_rest       
    }
      
    unicast_src_ip 192.168.1.100 
    unicast_peer {
        192.168.1.101
    }
   
    virtual_ipaddress {
        1192.168.1.103
    }
    
    notify_master    /opt/rest_status_master.sh
    notify_backup    /opt/rest_status_backup.sh
}
```

**/opt/rest_status_backup.sh**
```
#!/bin/bash
. /etc/profile
sleep 10
cp -f /etc/keepalived/keepalived.conf_withoutRest /etc/keepalived/keepalived.conf
systemctl reload keepalived
nohup java /opt/rest.jar >   /opt/rest.out 2>&1 &
```
while keepalived.conf_withoutRest is the same content but the vrrp_instance rest segment is removed

**expected behavior  log**
```
Jul 11 11:38:02 localhost systemd: Reloading LVS and VRRP High Availability Monitor.
Jul 11 11:38:02 localhost Keepalived[5327]: Reloading ...
Jul 11 11:38:02 localhost Keepalived[5327]: Opening file '/etc/keepalived/keepalived.conf'.
Jul 11 11:38:02 localhost Keepalived[5327]: Configuration file /etc/keepalived/keepalived.conf
Jul 11 11:38:02 localhost Keepalived[5327]: (Line 6) WARNING - number '0' outside range [0.000001, 4294.967295]
Jul 11 11:38:02 localhost Keepalived[5327]: (Line 6) vrrp_garp_interval '0' is invalid
Jul 11 11:38:02 localhost Keepalived[5327]: (Line 7) WARNING - number '0' outside range [0.000001, 4294.967295]
Jul 11 11:38:02 localhost Keepalived[5327]: (Line 7) vrrp_gna_interval '0' is invalid
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: Reloading
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: (/etc/keepalived/keepalived.conf: Line 6) WARNING - number '0' outside range [0.000001, 4294.967295]
Jul 11 11:38:02 localhost systemd: Reloaded LVS and VRRP High Availability Monitor.
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: (/etc/keepalived/keepalived.conf: Line 6) vrrp_garp_interval '0' is invalid
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: (/etc/keepalived/keepalived.conf: Line 7) WARNING - number '0' outside range [0.000001, 4294.967295]
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: (/etc/keepalived/keepalived.conf: Line 7) vrrp_gna_interval '0' is invalid
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: Script user 'keepalived_script' does not exist
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: (/etc/keepalived/keepalived.conf: Line 113) Truncating auth_pass to 8 characters
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: VRRP_Script(check_nginx) considered successful on reload
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: VRRP_Script(check_rest) considered successful on reload
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: SECURITY VIOLATION - scripts are being executed but script_security not enabled.
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: Assigned address 192.168.1.100 for interface eth0
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: (nginx) Changing effective priority from 100 to 120
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: (rest) sent 0 priority
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: (rest) removing VIPs.
Jul 11 11:38:02 localhost Keepalived[5327]: Reload complete
Jul 11 11:38:02 localhost Keepalived_vrrp[5331]: (nginx) setting VIPs.
```
**not expected behavior log**
**TBD**

**Comment by pqarmitage** [comment]
@SimonLiuhd Please provide your configuration and the keepalived logs so that we can see what is happening.

**Comment by pqarmitage** [comment]
@SimonLiuhd Could you please provide copies of your system logs, both for occasions when the behaviour is as expect, and also for when the behaviour is not as expected.

**Comment by SimonLiuhd** [comment]
@pqarmitage 
Thanks for your reply.
Added the expected behavior log in the original post.

I cannot reproduce the not expacted behavior now, but I am sure I saw it before.

While I think that when the reloading taking a long time on the machine A, which causes that the vrrp instances on the machine B does not recieved advertion for 3+ times (reaches the timeout), the vrrp instances on machine B will be promoted to master. so all other VIPs switch.

So the behavior depends on the reloading duration time：
when duration time less than timeout of advert, other VIPs are kept; otherwise, VIPs go away.
Is that correct?
